### PR TITLE
[TTPReq] TTP Awareness moment modal 

### LINKED ISSUE #14158: [TTPReq] TTP Awareness moment modal
Designs: pdfdoF-5KF-p2
Flows: pdfdoF-5GF-p2

## Description

1. Display an awareness modal once when an eligible user opens the app. Avoid showing it immediately after logging in to prevent overwhelming the user with too much new information. Instead, wait until the next app launch. Eligibility for this modal should be determined similarly to how the TTP badge visibility is managed on tabs and in the Payments menu.
2. Put the functionality under the feature flag
3. Enable Now button should close this modal and open Set up Tap to Pay flow.
4. Learn More button should close this modal and open About Tap to Pay flow.
5. Before JITM was used on the first launch, update the existing JITM to put the cap on the app version.

## Designs

Find the latest designs pdfdoF-5KF-p2 

Image

### PR DESCRIPTION
**Pull request by staskus**
Closes: #14158

## Description

Display an awareness modal to TTP-eligible users.

### View

Built a simple `TapToPayAwarenessMomentView` with a short title, description, image and two CTAs: Enable Now and Learn More. All taken from https://developer.apple.com/tap-to-pay/marketing-guidelines/#in-your-app.

### Navigation

I wanted to isolate the changes as much as possible within the awareness modal itself. I decided to use `DeepLinkNavigator` since it already knows how to navigate from the dashboard to the payments view.

'Enable Now' goes to Menu -> Payments -> Set up Tap to Pay
'Learn More' goes to Menu -> Payments -> About Tap to Pay

Does it make sense to you or is it better to avoid using the `DeepLinkNavigator` this way? 🤔 

### Presentation Logic

I created `TapToPayAwarenessMomentDeterminer` to contain the presentation logic. Eligibility criteria is similar to `TapToPayBadgePromotionChecker` that we show in Menu.

However, I've chosen to be a bit stricter, since this is a modal that appears on top of all the context:
- Show only once
- Show only to onboarded users
- Show only after the second launch (to avoid cases where it's presented right after user login)

I'm also thinking of disabling it for debug builds so it wouldn't affect us during testing sessions, what do you think?

We also have JITM logic on the backend for TTP. For now, I haven't changed anything related to it. Theoretically, they can both complement one another.

## Steps to reproduce

Prerequisites:
- Enable `tapToPayEducation` in `DefaultFeatureFlagService`
- Device and site eligible for TTP

### Presentation logic
1. Fresh install the app and log in
2. Confirm the modal doesn't appear
3. Go to Payments -> Menu and finish onboarding if necessary
4. Kill the app
5. Relaunch the app
6. Confirm the modal to try to tap to pay opens
7. Dismiss the modal
8. Kill & relaunch the app
9. Confirm the modal doesn't appear

### Modal logic

I recommend returning `true` from `shouldPresent()` so the app wouldn't need to be reinstalled

1. Launch the app & confirm the modal appears
2. Tap 'Enable Now'
3. Confirm the modal is closed and 'Set up Tap to Pay' view is opened
4. Relaunch the app & confirm the modal appears
5. Tap 'Learn more'
6. Confirm the modal is closed and 'About Tap to Pay' (Education) view is opened

## Testing information

Added `TapToPayAwarenessMomentDeterminerTests` tests for presentation logic, since this is where the most business logic is.

Tested on iPhone 14 Pro 17.7

## Videos

https://github.com/user-attachments/assets/954d72bf-a445-4073-9611-03d7d7d519d1

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [x] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [x] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [x] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on all devices (phone/tablet) and no regressions are added.

### CODE REVIEW

File: WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/Tap to Pay Education/TapToPayAwarenessMomentDeterminer.swift [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    private func setAttempted() {
+        userDefaults.set(true, forKey: Constants.firstAttemptKey)
+    }
+
+    private func isFirstAttempt() -> Bool {
+        !userDefaults.bool(forKey: Constants.firstAttemptKey)
+    }
+}
+
+extension TapToPayAwarenessMomentDeterminer {
```
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/Tap to Pay Education/TapToPayAwarenessMomentDeterminer.swift]: nit:
```suggestion
private extension TapToPayAwarenessMomentDeterminer {
```

File: WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/Tap to Pay Education/TapToPayAwarenessMomentDeterminer.swift [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    enum Constants {
+        static let previousPresentationKey = "TapToPayAwarenessMomentDeterminer.previousPresentationKey"
+        static let firstAttemptKey = "TapToPayAwarenessMomentDeterminer.firstAttemptKey"
+    }
```
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/Tap to Pay Education/TapToPayAwarenessMomentDeterminer.swift]: In the codebase, we generally define the keys in [this Key extension](https://github.com/woocommerce/woocommerce-ios/blob/11b514ac8b1ff179ca28fd32fbc3da1702517a4b/WooCommerce/Classes/Extensions/UserDefaults%2BWoo.swift#L8). Do you think it's worth keeping these user defaults keys there for consistency?

File: WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/Tap to Pay Education/TapToPayAwarenessMomentView.swift [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+                Image(imageName)
+                    .resizable()
+                    .scaledToFit()
+
+                Spacer()
+
+                Button(Localization.enable, action: {
+                    destination = PaymentsMenuDestination.tapToPay
+                    dismiss()
+
```
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/Tap to Pay Education/TapToPayAwarenessMomentView.swift]: super nit: empty line (same for L49)
```suggestion
```

File: WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/Tap to Pay Education/TapToPayAwarenessMomentView.swift [hunk 4]
```diff
@@ -0,0 +1,107 @@
+import Foundation
+import SwiftUI
+import Yosemite
+import WooFoundation
+
+struct TapToPayAwarenessMomentView: View {
```
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/Tap to Pay Education/TapToPayAwarenessMomentView.swift:6]: non-blocking suggestion: might SwiftUI preview be helpful for any potential future iterations?

File: WooCommerce/WooCommerceTests/Mocks/MockFeatureFlagService.swift [hunk 5]
```diff
@@ -1,30 +1,31 @@
...
-    private let googleAdsCampaignCreationOnWebView: Bool
-    private let blazeEvergreenCampaigns: Bool
-    private let blazeCampaignObjective: Bool
-    private let revampedShippingLabelCreation: Bool
-    private let viewEditCustomFieldsInProductsAndOrders: Bool
-    private let favoriteProducts: Bool
-    private let paymentsOnboardingInPointOfSale: Bool
-    private let isProductGlobalUniqueIdentifierSupported: Bool
-    private let isSendReceiptAfterPaymentEnabled: Bool
+class MockFeatureFlagService: FeatureFlagService {
```
Comment (jaclync) [inline WooCommerce/WooCommerceTests/Mocks/MockFeatureFlagService.swift]: nit:
```suggestion
final class MockFeatureFlagService: FeatureFlagService {
```

File: WooCommerce/WooCommerceTests/Mocks/MockFeatureFlagService.swift [hunk 6]
```diff
@@ -1,30 +1,31 @@
...
+    var googleAdsCampaignCreationOnWebView: Bool
+    var blazeEvergreenCampaigns: Bool
+    var blazeCampaignObjective: Bool
+    var revampedShippingLabelCreation: Bool
+    var viewEditCustomFieldsInProductsAndOrders: Bool
+    var favoriteProducts: Bool
+    var paymentsOnboardingInPointOfSale: Bool
+    var isProductGlobalUniqueIdentifierSupported: Bool
+    var isSendReceiptAfterPaymentEnabled: Bool
+    var tapToPayEducation: Bool = false
```
Comment (jaclync) [inline WooCommerce/WooCommerceTests/Mocks/MockFeatureFlagService.swift]: nit: maybe can use the default value from init for consistency?
```suggestion
    var tapToPayEducation: Bool
```

File: WooCommerce/Classes/ViewRelated/Dashboard/DashboardView.swift [hunk 7]
```diff
@@ -187,6 +187,9 @@ struct DashboardView: View {
         .sheet(isPresented: $viewModel.showingInAppFeedbackSurvey) {
             Survey(source: .inAppFeedback)
         }
+        .sheet(isPresented: $viewModel.showingTapToPayAwarenessMoment) {
+            TapToPayAwarenessMomentView()
+        }
```
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/Dashboard/DashboardView.swift:192]: I don't think we use it often, but there's also a remotely controlled modal [JITM that could be shown when loading the dashboard view](https://github.com/woocommerce/woocommerce-ios/blob/11b514ac8b1ff179ca28fd32fbc3da1702517a4b/WooCommerce/Classes/ViewRelated/Dashboard/DashboardViewHostingController.swift#L99-L114) in the UIKit way. Might be worth testing when both are presented by hard-coding the JITM modal.

**Review by jaclync — APPROVED** [review]
LGTM 🚢

File: WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/Tap to Pay Education/TapToPayAwarenessMomentDeterminer.swift (hunk 2 above)
Comment (staskus) [reply WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/Tap to Pay Education/TapToPayAwarenessMomentDeterminer.swift]: Thanks. Yes, I think it makes sense to move them there for consistency.

File: WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/Tap to Pay Education/TapToPayAwarenessMomentView.swift (hunk 4 above)
Comment (staskus) [reply WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/Tap to Pay Education/TapToPayAwarenessMomentView.swift:6]: I added previews. In most cases I develop SwiftUI views using [Injection](https://github.com/johnno1962/InjectionIII). I wish using previews would be as snappy.

File: WooCommerce/Classes/ViewRelated/Dashboard/DashboardView.swift (hunk 7 above)
Comment (staskus) [reply WooCommerce/Classes/ViewRelated/Dashboard/DashboardView.swift:192]: Thanks. Yes, I'm aware of JITM. I think we can disable TTP JITM remotely later for the version that this modal gets released. 

Optionally, we could consider keeping them both. Modal is quite intrusive and can get dismissed quickly while JITM just appears on the dashboard. 

I tested, this is how having both would look:

https://github.com/user-attachments/assets/c4debcea-2e41-475b-b759-59f14a582ee3

**Pull request merged**